src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -0,0 +1,180 @@
+import _isEmpty from 'lodash/isEmpty';
+import _isObject from 'lodash/isObject';
+import _map from 'lodash/map';
+import LegModel from './LegModel';
+import { routeStringFormatHelper } from '../../navigationLibrary/Route/routeStringFormatHelper';
+
+/**
+ *
+ * This class should always be instantiated from an `AircraftInstanceModel` and
+ * always instantiated with some form of a `spawnPatternModel`.
+ *
+ * @class Fms
+ */
+export default class Fms {
+    /**
+     * @constructor
+     * @param aircraftInitProps {object}
+     * @param initialRunwayAssignment {string}
+     * @param navigationLibrary {NavigationLibrary}
+     */
+    constructor(aircraftInitProps, initialRunwayAssignment, navigationLibrary) {
+        if (!_isObject(aircraftInitProps) || _isEmpty(aircraftInitProps)) {
+            throw new TypeError('Invalid aircraftInitProps passed to Fms');
+        }
+
+        /**
+         *
+         *
+         * @property _navigationLibrary
+         * @type {NavigationLibrary}
+         * @private
+         */
+        this._navigationLibrary = navigationLibrary;
+
+        /**
+         *
+         *
+         * @property _runway
+         * @type {string}
+         * @private
+         */
+        this._runway = initialRunwayAssignment;
+
+        /**
+         *
+         *
+         * @property legCollection
+         * @type {array}
+         * @default []
+         */
+        this.legCollection = [];
+
+        /**
+         *
+         *
+         * @property category
+         * @type {string}
+         * @default ''
+         */
+        this.category = '';
```

Comment:
Well we can kinda deduce whether the aircraft is landing based on whether the last thing in their flightplan is the ICAO code of an airport, so the category isn't too necessary. The `runway` however will be needed to be known by the aircraft for a myriad of things.